pipeline/lib/metric.ml

```diff
@@ -0,0 +1,134 @@
+module Github = Current_github
+module BenchmarksData = Map.Make (String)
+
+let parse_benchmark_data ~data =
+  Array.fold_left
+    (fun acc datum ->
+      match datum with
+      | [| _commit; benchmark_name; test_name; metrics |] ->
+          let key = Fmt.str "%s__%s" benchmark_name test_name in
+          let parsed_metrics = Yojson.Safe.from_string metrics in
+          BenchmarksData.add key parsed_metrics acc
+      | _ -> acc)
+    BenchmarksData.empty data
+
+let sum_float_list xs = List.fold_left (fun acc x -> acc +. x) 0. xs
+let calc_diff v cv = Float.abs (v -. cv) /. cv *. 100.
+
+let find_avg vs =
+  vs
+  |> List.filter_map (function
+       | key, value when key = "avg" -> Some value
+       | _ -> None)
+  |> function
+  | [ avg ] -> avg
+  | _ -> 0.
+
+let change_threshold = 0.01
+
+let find_changed_metrics ~metrics ~compare_metrics =
+  BenchmarksData.mapi
+    (fun key benchmark ->
+      let compare_benchmark = BenchmarksData.find_opt key compare_metrics in
+      match compare_benchmark with
+      | Some compare_benchmark ->
+          let compare_benchmark_metrics =
+            compare_benchmark
+            |> Current_bench_json.Latest.metrics_of_json []
+            |> List.fold_left
+                 (fun acc
+                      ({ name; value; _ } : Current_bench_json.Latest.metric) ->
+                   BenchmarksData.add name value acc)
+                 BenchmarksData.empty
+          in
+          benchmark
+          |> Current_bench_json.Latest.metrics_of_json []
+          |> List.fold_left
+               (fun acc ({ name; value; _ } : Current_bench_json.Latest.metric) ->
+                 let compare_value =
+                   BenchmarksData.find_opt name compare_benchmark_metrics
+                 in
+                 let delta =
+                   match compare_value with
+                   | Some compare_value -> (
+                       match (value, compare_value) with
+                       | Float v, Float cv when cv != 0. -> calc_diff v cv
+                       | Floats vs, Floats cvs ->
+                           let v = sum_float_list vs in
+                           let cv = sum_float_list cvs in
```

Comment:
Ah I think the current meaning of `Floats` is "repeated benchmarks" (so comparing the sum without dividing by the length might cause some issues if the user repeats the benchmark more in the future)

Reply:
Ah, yes. I think using the mean is a safe way to do this. Thanks for catching this!